Sources/Sentry/SentryScope.m

```diff
@@ -86,17 +86,8 @@ - (instancetype)initWithScope:(SentryScope *)scope
     if (self = [self init]) {
         self.extraDictionary = scope.extraDictionary.mutableCopy;
         self.tagDictionary = scope.tagDictionary.mutableCopy;
-        SentryUser *scopeUser = scope.userObject;
-        SentryUser *user = nil;
-        if (nil != scopeUser) {
-            user = [[SentryUser alloc] init];
-            user.userId = scopeUser.userId;
-            user.data = scopeUser.data.mutableCopy;
-            user.username = scopeUser.username;
-            user.email = scopeUser.email;
-        }
         self.maxBreadcrumbs = scope.maxBreadcrumbs;
-        self.userObject = user;
+        self.userObject = scope.userObject.copy;
```

Comment:
Is this a deep copy? because the deleted code had: `scopeUser.data.mutableCopy` so maybe a unit test that makes sure those are all different references would help

Reply:
Yes, this is a deep copy as `SentryUser` has `NSCopying` implemented, see
https://github.com/getsentry/sentry-cocoa/blob/54f31ec920badd985f7b6dc8c8c2c7401db75cec/Sources/Sentry/SentryUser.m#L22-L35. It is basically the same code, that's why I got rid of it. I'm going to add a test for it to make sure.